Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/TimeOffController.cs

```diff
@@ -244,29 +244,52 @@ internal async Task ProcessTimeOffsAsync(string isRequestFromLogicApp)
             List<PayCodeToTimeOffReasonsMappingEntity> timeOffRequestsPayCodeList,
             List<GlobalTimeOffRequestItem> globalTimeOffRequestDetails)
         {
+            this.telemetryClient.TrackTrace($"ProcessTimeOffEntitiesBatchAsync started.");
+
             foreach (var item in processKronosUsersQueueInBatch)
             {
                 foreach (var timeOffRequestItem in timeOffResponseDetails.Where(x => x.Employee.PersonIdentity.PersonNumber == item.KronosPersonNumber))
                 {
                     if (timeOffRequestItem.StatusName.Equals(ApiConstants.ApprovedStatus, StringComparison.Ordinal))
                     {
+                        this.telemetryClient.TrackTrace($"ProcessTimeOffEntitiesBatchAsync look up count: {lookUpData.Count} ");
+
                         if (lookUpData.Count == 0)
                         {
+                            // Performing the query using the RowKey as the paycode name from Kronos; and the PartitionKey as the TeamID.
```

Comment:
In most cases, the comment should describe at a conceptual level what you're doing. The specific mechanics is often clear from the code. If you just repeat what the code is already doing, you risk being inconsistent with it, as here, where your comment of "Performing the query ... RowKey ... PartitionKey" makes the reader think there's a query to table store, when really it's just a linear scan of a List.

(Which is inefficient, BTW-- why is that not a dictionary?)

Reply:
@aosolis I have updated the comments with your guidance, and I agree that the dictionary data structure is more appropriate than using a list. However, we are returning the list from our database, and not converting that into key value pairs as the time off reasons are scoped to a particular team.